oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/JwtIssuerValidatorTests.java

```diff
@@ -86,16 +130,37 @@ public void validateWhenIssuerMatchesAndIsNotAUriThenReturnsSuccess() {
 		// @formatter:on
 	}
 
+	// gh-6073
+	@Test
+	public void validateWhenAllowEmptyFalseAndIssuerMatchesAndIsNotAUriThenReturnsSuccess() {
+		Jwt jwt = TestJwts.jwt().claim(JwtClaimNames.ISS, "issuer").build();
+		JwtIssuerValidator validator = new JwtIssuerValidator("issuer");
+		validator.setAllowEmpty(false);
+		// @formatter:off
+		assertThat(validator.validate(jwt))
+				.isEqualTo(OAuth2TokenValidatorResult.success());
+		// @formatter:on
+	}
+
+	@Test
+	public void validateWhenAllowEmptyDefaultAndJwtIsNullThenThrowsIllegalArgumentException() {
```

Comment:
Thank you for these additional tests. Will you please place any tests or other changes that are unrelated to the feature in separate commits so that each commit is atomic?